<a href="https://colab.research.google.com/github/ankitsrm/MachineLearning/blob/main/Understanding_DATA_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

df = pd.read_csv('train.csv')

***How big is your data ? :***

In [2]:
df.shape

(891, 12)

***How does the data look like ? ***

In [3]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Some time we also see BIAS -- which means sure wale rows kuch similar hai bad wale rows bhi similar hai

Where we dont want to see the BIAS ?? To see that randomly data rows

In [5]:
df.sample(5)

PassengerId  Survived  Pclass                             Name   Sex  \
287          288         0       3             Naidenoff, Mr. Penko  male   
420          421         0       3           Gheorgheff, Mr. Stanio  male   
728          729         0       2  Bryhl, Mr. Kurt Arnold Gottfrid  male   
597          598         0       3              Johnson, Mr. Alfred  male   
225          226         0       3     Berglund, Mr. Karl Ivar Sven  male   

      Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
287  22.0      0      0   349206   7.8958   NaN        S  
420   NaN      0      0   349254   7.8958   NaN        C  
728  25.0      1      0   236853  26.0000   NaN        S  
597  49.0      0      0     LINE   0.0000   NaN        S  
225  22.0      0      0  PP 4348   9.3500   NaN        S

***What is the data type of cols ? ***

it will tell the most information of the data

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


***Are there any missing value ??***

ek way hai above in the info :
 in df.info()
age have 741 and cabin have 204 so there is a missing value

In [8]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

***How does the data look mathematically ? ***




`df.describe()` gives mathematicall value for this

In [9]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

***Are there Duplicate values  ? ***

In [10]:
df.duplicated().sum()

0

***How is the correlation between cols ?***

Konsa input column jo output ke uper koi effect nahi daal rha hai usko we remove it

In [11]:
df.corr()

<ipython-input-11-2f6f6606aa2c>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.005007 -0.035144  0.036847 -0.057527 -0.001652   
Survived       -0.005007  1.000000 -0.338481 -0.077221 -0.035322  0.081629   
Pclass         -0.035144 -0.338481  1.000000 -0.369226  0.083081  0.018443   
Age             0.036847 -0.077221 -0.369226  1.000000 -0.308247 -0.189119   
SibSp          -0.057527 -0.035322  0.083081 -0.308247  1.000000  0.414838   
Parch          -0.001652  0.081629  0.018443 -0.189119  0.414838  1.000000   
Fare            0.012658  0.257307 -0.549500  0.096067  0.159651  0.216225   

                 Fare  
PassengerId  0.012658  
Survived     0.257307  
Pclass      -0.549500  
Age          0.096067  
SibSp        0.159651  
Parch        0.216225  
Fare         1.000000

Isme value -1 se 1 ke bitch mai hota hai jitna negative value hai to inversaly propostional and positive value hai to directly proportional

For example mera output column survived hai to i want to find the correlation between survived with all other input column

In [12]:
df.corr()['Survived']

<ipython-input-12-57d70bb92b5b>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()['Survived']


PassengerId   -0.005007
Survived       1.000000
Pclass        -0.338481
Age           -0.077221
SibSp         -0.035322
Parch          0.081629
Fare           0.257307
Name: Survived, dtype: float64

so we can see here survived ka survived ke sath correlation is one. which it will be but

1.   Fare ka chance survived kaa match jada hai .25
2.   Same goes with pClass -0.33 so it will reduce survival.
3.   PassengerID -- -0.005 will not give anything
